baseplate/lib/prometheus_metrics.py

```diff
@@ -17,6 +18,146 @@
+    "http_server_request_size_bytes",
+    "Description of histogram",
+    http_server_histogram_labels,
+    buckets=default_buckets,
+)
+http_server_response_size_bytes = Histogram(
+    "http_server_response_size_bytes",
+    "Description of histogram",
+    http_server_histogram_labels,
+    buckets=default_buckets,
+)
+http_server_requests_total = Counter(
+    "http_server_requests_total",
+    "Description of counter",
+    http_server_requests_total_labels,
+)
+http_server_active_requests = Gauge(
+    "http_server_active_requests",
+    "Description of gauge",
+    http_server_active_requests_labels,
+)
+
+
+# http client labels and metrics
+http_client_latency_labels = [
+    "http_method",
+    "http_success",
+    "http_endpoint",
+    "http_slug",
+]
+http_client_requests_total_labels = [
+    "http_method",
+    "http_success",
+    "http_endpoint",
+    "http_response_code",
+    "http_slug",
+]
+http_client_active_requests_labels = [
+    "http_method",
+    "http_endpoint",
+    "http_slug",
+]
+
+http_client_latency_seconds = Histogram(
+    "http_client_latency_seconds",
+    "Description of histogram",
+    http_client_latency_labels,
+    buckets=default_buckets,
+)
+http_client_requests_total = Counter(
+    "http_client_requests_total",
+    "Description of counter",
+    http_client_requests_total_labels,
+)
+http_client_active_requests = Gauge(
+    "http_client_active_requests",
+    "Description of gauge",
+    http_client_active_requests_labels,
+)
+
+
+class PrometheusHTTPClientMetrics:
+    def __init__(self) -> None:
+        pass
+
+    def latency_seconds_metric(self, tags: Dict) -> Histogram:
+        return http_client_latency_seconds.labels(
+            http_method=tags.get("http.method", ""),
+            http_endpoint=getHTTPEndpointHandler(tags.get("http.url", "")),
+            http_success=tags.get("http.success", ""),
+            http_slug=tags.get("http.slug", ""),
+        )
+
+    def requests_total_metric(self, tags: Dict) -> Counter:
+        return http_client_requests_total.labels(
+            http_method=tags.get("http.method", ""),
+            http_endpoint=getHTTPEndpointHandler(tags.get("http.url", "")),
+            http_success=tags.get("http.success", ""),
+            http_response_code=tags.get("http.status_code", ""),
+            http_slug=tags.get("http.slug", ""),
+        )
+
+    def active_requests_metric(self, tags: Dict) -> Gauge:
+        return http_client_active_requests.labels(
+            http_method=tags.get("http.method", ""),
+            http_endpoint=getHTTPEndpointHandler(tags.get("http.url", "")),
+            http_slug=tags.get("http.slug", ""),
+        )
+
+    def get_latency_seconds_metric(self) -> Histogram:
+        return http_client_latency_seconds
+
+    def get_requests_total_metric(self) -> Counter:
+        return http_client_requests_total
+
+    def get_active_requests_metric(self) -> Gauge:
+        return http_client_active_requests
```

Comment:
The bare one should mostly not be used I guess?
It's potentially useful in tests (where you get to access the underlying object and access all the metrics at once, independent of there labels), or if you're trying to do something very custom like exposing these with labels coming from something else than `tags`?

Reply:
Test use case makes sense, I see that's how we're doing it for Thrift. Would usually prefer an implementation detail like this be hidden or named explicitly `get_active_requests_metric[_for_testing]` but since this isn't intended to be used directly by most developers I think it's fine 👍